### PR TITLE
Adde xTB and xTB-GSM

### PR DESCRIPTION
**Pull request by alongd**
Added the xTB and xTB-GSM Job adapters.
This PR also includes an infrastructure for running these incore adapters, e.g., an internal ARC YAML format was implemented.

Note that torchANI is also added in this PR, yet knocked-out in subsequent commits due to environment issues. It's recommended to re-implement it and execute it as a sub-process in its own environment, utilizing the YAML format to store the output.

### CODE REVIEW

File: arc/common.py [hunk 1]
```diff
@@ -1589,3 +1589,30 @@ def _check_r_n_p_symbols_between_rmg_and_arc_rxns(arc_reaction: 'ARCReaction',
...
     return result
+
+
+def safe_copy_file(source: str,
+                   destination: str,
+                   wait: int = 10,
+                   max_cycles: int = 100,
+                   ):
+    """
+    Copt a file safely.
```
Comment (kfir4444) [inline arc/common.py:1600]: Typo?

File: Makefile [hunk 2]
```diff
@@ -24,6 +24,16 @@ install-gcn-cpu:
 install-kinbot:
 	bash devtools/install_kinbot.sh
 
+install-sella:
+	bash devtools/install_sella.sh
+
+install-xtb:
+	bash devtools/install_xtb.sh
+
+# TorchANI is included in the arc_env, install_torchani.sh is not executed by ``make install-all``
```
Comment (kfir4444) [inline Makefile:33]: Does ARC env include torchani after the change?

File: arc/job/adapters/__init__.py [hunk 3]
```diff
@@ -4,4 +4,6 @@
 import arc.job.adapters.orca
 import arc.job.adapters.qchem
 import arc.job.adapters.terachem
+# import arc.job.adapters.torch_ani
```
Comment (kfir4444) [inline arc/job/adapters/__init__.py:7]: Can you clear this comment?

File: arc/job/adapters/gaussian.py [hunk 4]
```diff
@@ -396,7 +396,7 @@ def execute_incore(self):
               raise_msg=f'Please install {self.job_adapter}, see {self.url} for more information.',
               )
         self._log_job_execution()
-        execute_command(incore_commands[self.server][self.job_adapter])
+        execute_command(incore_commands[self.job_adapter])
```
Comment (kfir4444) [inline arc/job/adapters/gaussian.py:399]: Great fix!

File: arc/job/adapters/xtb_adapter.py [hunk 5]
```diff
@@ -0,0 +1,439 @@
+"""
```
Comment (kfir4444) [inline arc/job/adapters/xtb_adapter.py:1]: Normally, the name of adapters is the `{software}.py`. For consistency, please rename.

File: devtools/install_torchani.sh [hunk 6]
```diff
@@ -0,0 +1,10 @@
+# Properly configure the shell to use 'conda activate'.
```
Comment (kfir4444) [inline devtools/install_torchani.sh:1]: I wonder if installing torchani from the install file could have solved the issue we had before. Might worth a shot when we get back to it.

File: devtools/install_kinbot.sh [hunk 7]
```diff
@@ -1,3 +1,7 @@
+# Properly configure the shell to use 'conda activate'.
```
Comment (kfir4444) [inline devtools/install_kinbot.sh:1]: Another way could be `conda init bash`

**Review by kfir4444 — CHANGES_REQUESTED** [review]
Hey @alongd, great job!
There are a few fixes, please see comments.
Note that although torchani is not included in this PR, there is a lot of preparations for it, I think you should maybe open an issue about it, with links to the relevant branches/commits, so that we will remember how to restart it if and when some time passes.

**Comment by alongd** [comment]
@kfir4444, all changes were performed, feel free to merge unless there are additional comments

**Review by kfir4444 — APPROVED** [review]

**Pull request merged**